notificationworkerlambda/src/test/scala/com/gu/notifications/worker/WorkerRequestHandlerSpec.scala

```diff
@@ -0,0 +1,140 @@
+      sendingResults shouldEqual Some(SendingResults(1, 0, 0))
+      tokensToCleanCount shouldEqual 0
+    }
+
+    "Clean invalid tokens" in new WRHSScope {
+      override def deliveries: Stream[IO, Either[DeliveryException, ApnsDeliverySuccess]] =
+        Stream(
+          Left(InvalidToken(UUID.randomUUID, "invalid token", "test"))
+        )
+
+      workerRequestHandler.handleRequest(sqsEvent, null)
+
+      deliveryCallsCount shouldEqual 1
+      cloudwatchCallsCount shouldEqual 1
+      cleaningCallsCount shouldEqual 1
+      sendingResults shouldEqual Some(SendingResults(0, 1, 0))
+      tokensToCleanCount shouldEqual 1
+    }
+
+    "Count dry runs" in new WRHSScope {
+      override def deliveries: Stream[IO, Either[DeliveryException, ApnsDeliverySuccess]] =
+        Stream(
+          Right(ApnsDeliverySuccess("token", dryRun = true))
+        )
+
+      workerRequestHandler.handleRequest(sqsEvent, null)
+
+      deliveryCallsCount shouldEqual 1
+      cloudwatchCallsCount shouldEqual 1
+      cleaningCallsCount shouldEqual 1
+      sendingResults shouldEqual Some(SendingResults(0, 0, 1))
+      tokensToCleanCount shouldEqual 0
+    }
+  }
+
+
+  trait WRHSScope extends Scope  {
+
+    val notification = BreakingNewsNotification(
+      id = UUID.fromString("068b3d2b-dc9d-482b-a1c9-bd0f5dd8ebd7"),
+      `type` = NotificationType.BreakingNews,
+      title  = "French president Francois Hollande says killers of Normandy priest claimed to be from Islamic State",
+      message = "French president Francois Hollande says killers of Normandy priest claimed to be from Islamic State",
+      thumbnailUrl = None,
+      sender = "matt.wells@guardian.co.uk",
+      link = Internal("world/2016/jul/26/men-hostages-french-church-police-normandy-saint-etienne-du-rouvray", Some("https://gu.com/p/4p7xt"), GITContent),
+      imageUrl = None,
+      importance = Major,
+      topic = List(Topic(Breaking, "uk"), Topic(Breaking, "us"), Topic(Breaking, "au"), Topic(Breaking, "international"))
+    )
+    val shardedNotification = ShardedNotification(
+      notification = notification,
+      range = ShardRange(0, 1)
+    )
+
+    val sqsEvent: SQSEvent = {
+      val event = new SQSEvent()
+      val sqsMessage = new SQSMessage()
+      sqsMessage.setBody(Json.stringify(Json.toJson(shardedNotification)))
+      event.setRecords(List(sqsMessage).asJava)
+      event
+    }
+
+
+    def deliveries: Stream[IO, Either[DeliveryException, ApnsDeliverySuccess]] = Stream(Right(ApnsDeliverySuccess("token")))
+
+    var deliveryCallsCount = 0
+    var cleaningCallsCount = 0
+    var tokensToCleanCount = 0
+    var cloudwatchCallsCount = 0
+    var sendingResults: Option[SendingResults] = None
+
+    val workerRequestHandler = new WorkerRequestHandler[ApnsClient] {
+      override def platform: Platform = iOS
+
+      override def deliveryService: IO[DeliveryService[IO, ApnsClient]] = IO.pure(new DeliveryService[IO, ApnsClient] {
```

Comment:
Do you need override?

Reply:
Technically I don't think the keyword override is necessary, it just happen it's included automatically by intelliJ. It also has the nice effect that if the signature changes, or the attribute is dropped, I'll get a compilation error, whereas dropping the keyword won't catch it.